src/hotspot/share/gc/g1/g1Policy.hpp

```diff
@@ -301,7 +301,8 @@ class G1Policy: public CHeapObj<mtGC> {
 
   void init(G1CollectedHeap* g1h, G1CollectionSet* collection_set);
 
-  void note_gc_start();
+  void note_young_gc_pause_start();
+  void note_young_gc_pause_end();
```

Comment:
It's not obvious to me if there's a diff btw `young_gc_pause` and  `young_collection`, or what the diff is. I think having some comments for these two pairs of methods would make it easier to understand.

Reply:
Added comments. `gc_pause` is the whole pause, `collection` is about just the collection part (as defined by the inner scope of the `do_collection_pause_at_safepoint_helper`). Feel free to suggest better naming.

Reply:
Maybe something like `record_young_collection_work_start`; another question is whether we should keep the `note_` prefix for the pause or change it to `record_` as well as it seems quite common.

Reply:
I suggest we change to record_